src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -344,9 +347,10 @@ OMapInnerNode::merge_entry(omap_context_t oc, internal_iterator_t iter, OMapNode
       logger().debug("{}::merge_entry balanced l {} r {}", __func__, *l, *r);
       return l->make_balanced(oc, r).safe_then([=] (auto tuple) {
         auto [replacement_l, replacement_r, replacement_pivot] = tuple;
-        //update will not cuase overflow, do it first
+        //update operation will not cuase node overflow, so we can do it first
         journal_inner_update(liter, replacement_l->get_laddr(), maybe_get_delta_buffer());
-        if (!extent_will_overflow(replacement_pivot.size() + 1, std::nullopt)) {
+        bool overflow = extent_will_overflow(replacement_pivot.size() + 1, std::nullopt);
+        if (!overflow) {
           journal_inner_replace(riter, replacement_r->get_laddr(),
                                 replacement_pivot, maybe_get_delta_buffer());
           std::list<laddr_t> dec_laddrs{l->get_laddr(), r->get_laddr()};
```

Comment:
I called this out in earlier review;  what is the motivation for preferring std::list?  I suspect not access to push_front()--this can usually be avoided;  if really not avoidable, we could use a std::deque...

Reply:
@adamemerson ^^

Reply:
what do you prefer? vector?

Reply:
yes, from everything I can tell, this is a small concrete sequence, so choose vector

Reply:
updated.